shuffle/cli/tests/common/mod.rs

```diff
@@ -6,50 +6,67 @@ use diem_sdk::{
     crypto::ed25519::Ed25519PrivateKey, transaction_builder::TransactionFactory,
     types::LocalAccount,
 };
-use forge::ChainInfo;
+use forge::{AdminContext, ChainInfo};
 use shuffle::{
     account, deploy,
     dev_api_client::DevApiClient,
-    new,
-    shared::{self, Home, Network, NetworkHome},
+    new, shared,
+    shared::{Home, Network, NetworkHome, NetworksConfig},
 };
-use std::{convert::TryFrom, path::PathBuf, str::FromStr};
-use tempfile::TempDir;
+use smoke_test::scripts_and_modules::enable_open_publishing;
+use std::{
+    collections::BTreeMap,
+    convert::TryFrom,
+    fs,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+use tempfile::tempdir;
+use tokio::runtime::Runtime;
 use url::Url;
 
+const FORGE_NETWORK_NAME: &str = "forge";
+
+#[allow(dead_code)]
 pub struct ShuffleTestHelper {
     home: Home,
     network: Network,
+    networks_config: NetworksConfig,
     network_home: NetworkHome,
-    tmp_dir: TempDir,
+    home_path: PathBuf,
```

Comment:
We need the `tmp_dir` to remain on the `struct` because otherwise the underlying `dir` gets deleted when it's out of scope due to the `Drop`. Let's make `home_path()` a method and keep `tmp_dir`.

https://docs.rs/tempfile/3.0.2/tempfile/struct.TempDir.html